### PR TITLE
Add docker generation to Github Actions

### LINKED ISSUE #574: TODO: Build CI container from GitHub Actions
**Requested feature**

Add a GitHub Action workflow to build CI Docker Image.

**Motivation**

Make refreshing CI image easier and more repeatable by replacing the manual process. Include some best practices like image signing.

### PR DESCRIPTION
**Pull request by mathias-arm**
Fixes #574. Add image signature and verification based on cosign

### CODE REVIEW

**Review by egrimley-arm — DISMISSED** [review]
I don't see any problems, but I can't claim to understand everything in the `.yml` files. I wonder if any instructions should be added, though, so that people know how to update things in future?

**Comment by mathias-arm** [comment]
> I don't see any problems, but I can't claim to understand everything in the `.yml` files. I wonder if any instructions should be added, though, so that people know how to update things in future?

I added some instructions.

File: documents/dev/CI_PROCESSES.markdown [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+```
+COSIGN_EXPERIMENTAL=true cosign sign -y \
+	ghcr.io/veracruz-project/veracruzci@sha256:b5eb834cd58c69e6acc19b1b19c6345a020ad2ad090cdab3bc9f7d5d48fb00ba
+```
+
+4) The Digest of the Docker image in the project packages registry should match the value found in the logs (e.g. `sha256:b5eb834cd58c69e6acc19b1b19c6345a020ad2ad090cdab3bc9f7d5d48fb00ba`). This value should now be used to edit `.github/workflows/main.yml` and replace the value of the previous image. This change should be commited and pushed to the branch.
+
+5) A pull request for the branch should now be created. This will trigger the execution of CI using the new Docker image.
+
+6) If CI is successful and the pull request is acceptable, then the pull should be merged and the branch deleted. The commit from which the image was built should be preserved and be part of branch that is merged. If CI is not successful and changes need to be made to the Docker files. The commit modifying `github/workflows/main.yml` should be backed out, and the process should resume from step 3.
```
Comment (egrimley-arm) [inline documents/dev/CI_PROCESSES.markdown:20]: s/files. The/files, the/

**Review by egrimley-arm — DISMISSED** [review]
The documentation seems very clear. Thanks. I've suggested a cosmetic change to the last line of the file if you can be bothered.

**Review by egrimley-arm — APPROVED** [review]

**Pull request merged**